datalake_common/tests/test_record.py

```diff
@@ -76,11 +76,13 @@ def test_no_such_bucket(s3_connection):
 def test_no_end(random_metadata, s3_file_from_metadata):
     url = 's3://foo/baz'
     del(random_metadata['end'])
+    expected_metadata = random_metadata.copy()
```

Comment:
if all metadata has to have an end, shouldn't `random_metadata`, too?

Reply:
The intent here was to test the case where the stored metadata has no `end` key since:
1) legacy metadata blobs that are already in s3 may not have an `end` key 
2) even if we backfill end keys, some users and clients may still have ways to create new metadata blobs with no `end` (e.g. directly editing metadata in s3)

When the metadata is missing an `end` key, we expect datalake metadata read tools to fill in a default `end` key with a value of `None`